astropy/io/fits/fitsrec.py

```diff
@@ -478,9 +478,22 @@ def from_columns(cls, columns, nrows=0, fill=False, character_as_bytes=False):
         return data
 
     def __repr__(self):
-        # Force use of the normal ndarray repr (rather than the new
-        # one added for recarray in Numpy 1.10) for backwards compat
-        return np.ndarray.__repr__(self)
+        repr_dtype = self.dtype
+        # FIXME: remove numpy.record in the dtype repr, as recarray?
+        # if repr_dtype.type is np.record:
+        #     repr_dtype = np.dtype((np.void, repr_dtype))
+        prefix = "FITS_rec("
+        fmt = "FITS_rec(%s,%sdtype=%s)"
+        # get data/shape string. logic taken from numeric.array_repr
```

Comment:
This sent me to the wrong place, but I now realize the comment also comes from recarray - hopefully that will now be clearer with the comment I suggested on top.